src/shogun/features/DenseFeatures.cpp

```diff
@@ -265,23 +267,51 @@ template<class ST> void CDenseFeatures<ST>::feature_subset(int32_t* idx, int32_t
 	}
 }
 
-template<class ST> SGMatrix<ST> CDenseFeatures<ST>::get_feature_matrix()
+template <class ST>
+SGMatrix<ST> CDenseFeatures<ST>::get_feature_matrix()
 {
 	if (!m_subset_stack->has_subsets())
 		return feature_matrix;
 
-	SGMatrix<ST> submatrix(num_features, get_num_vectors());
+	SGMatrix<ST> target(num_features, get_num_vectors());
+	copy_feature_matrix(target);
+	return target;
+}
+
+template <class ST>
+void CDenseFeatures<ST>::copy_feature_matrix(SGMatrix<ST> target, index_t column_offset) const
+{
+	REQUIRE(column_offset>=0, "Column offset (%d) cannot be negative!\n", column_offset);
 
-	/* copy a subset vector wise */
-	for (int32_t i=0; i<submatrix.num_cols; ++i)
+	index_t num_vecs=get_num_vectors();
+	index_t num_cols=num_vecs+column_offset;
+
+	REQUIRE(target.matrix!=nullptr, "Provided matrix is not allocated!\n");
+	REQUIRE(target.num_rows==num_features,
+			"Number of rows (%d for given matrix) should be equal to the number of features (%d)!\n",
+			target.num_rows, num_features);
+	REQUIRE(target.num_cols>=num_cols,
+			"Number of cols (%d for given matrix) should be at least %d!\n",
+			target.num_cols, num_cols);
+
+	if (!m_subset_stack->has_subsets())
 	{
-		int32_t real_i = m_subset_stack->subset_idx_conversion(i);
-		memcpy(&submatrix.matrix[i*int64_t(num_features)],
-				&feature_matrix.matrix[real_i * int64_t(num_features)],
-				num_features * sizeof(ST));
+		auto input_begin=feature_matrix.matrix;
+		auto input_end=input_begin+feature_matrix.size();
+		auto output_begin=target.matrix+int64_t(num_features)*column_offset;
+		std::copy(input_begin, input_end, output_begin);
```

Comment:
should replace with newly added `shogun::memcpy` here. Commenting just so I don't forget